src/components/ifxRange/ifxRange.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-range',
+  styleUrl: 'ifxRange.scss',
+  shadow: true,
+})
+export class IfxRange {
+  @Prop() min: number = 0;
+  @Prop() max: number = 100;
+  @Prop({ mutable: true }) value: number = 0;
+  @Prop() disabled: boolean = false;
+  @Prop() showPercentage: boolean = false; // New property for toggleable percentage display
+  @Prop() leftIcon: string;
+  @Prop() rightIcon: string;
+  @Prop() leftText: string;
+  @Prop() rightText: string;
+  @Event() valueChanged: EventEmitter<number>;
+
+  private inputRef: HTMLInputElement;
+
+  handleInputChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.value = parseInt(target.value);
+    this.valueChanged.emit(this.value);
+    this.updateValuePercent();
+  }
+
+  updateValuePercent() {
+    if (this.inputRef) {
+      const percent = ((this.value - this.min) / (this.max - this.min)) * 100;
+      this.inputRef.style.setProperty('--value-percent', `${percent}%`);
+    }
+  }
+
+  componentDidLoad() {
+    this.updateValuePercent();
+  }
+
+
+  render() {
+    return (
+      <div class="ifx-range">
+        {this.leftText && (
```

Comment:
Icon + Text variants should not be allowed at the same time. When one is enabled, the other one should
be disabled.

Reply:
I don't think this should be the case. they work well together, no?

Reply:
That's the design in Figma:
<img width="257" alt="range" src="https://user-images.githubusercontent.com/108734092/233100637-9f9938d9-2d49-4c7e-a5de-3c3c1a79bc92.PNG">